launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/CreateByLinkServlet.java

```diff
@@ -0,0 +1,147 @@
+
+  /**
+  * Helper method for repeated code in catching exceptions.
+  */
+  private void writeResponse(HttpServletResponse res, String message, int statusCode) throws IOException {
+    res.setContentType("text/html");
+    res.getWriter().println(message);
+    res.setStatus(statusCode);
+  }
+
+  /**
+   * Requests user inputs in form fields, then creates Entity and places in Datastore.
+   * @throws IOException,IllegalArgumentException
+   */
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws IllegalArgumentException, IOException {
+    String episodeTitle = req.getParameter(EPISODE_TITLE);
+    String episodeDescription = req.getParameter(EPISODE_DESCRIPTION);
+    String episodeLanguage = req.getParameter(EPISODE_LANGUAGE);
+    String mp3Link = req.getParameter(MP3_LINK);
+    String id = req.getParameter(ID);
+
+    UserService userService = UserServiceFactory.getUserService();
+    String email = "";
+    if (userService.isUserLoggedIn()) {
+      email = userService.getCurrentUser().getEmail();
+    }
+
+    if (Strings.isNullOrEmpty(episodeTitle)) {
+      throw new IllegalArgumentException("No episode title inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(episodeDescription)) {
+      throw new IllegalArgumentException("No episode description inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(episodeLanguage)) {
+      throw new IllegalArgumentException("No episode language inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(mp3Link)) {
+      throw new IllegalArgumentException("No mp3 link inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(id)) {
+      throw new IllegalArgumentException("Sorry, no entity Id could be found.");
+    } else if (Strings.isNullOrEmpty(email)) {
+      throw new IllegalArgumentException("You are not logged in. Please try again.");
+    }
+
+    Key entityKey = null;
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity desiredFeedEntity;
+    try {
+      // Use key to retrieve entity from Datastore
+      entityKey = KeyFactory.stringToKey(id);
+      desiredFeedEntity = datastore.get(entityKey);
+    } catch (IllegalArgumentException e) {
+      // If entityId cannot be converted into a key
+      writeResponse(res, "Sorry, this is not a valid id.", HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    } catch (EntityNotFoundException e) {
+      // No matching entity in Datastore
+      writeResponse(res, "Your entity could not be found.", HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+
+    // Modify the xml string
+    RSS rssFeed = XML_MAPPER.readValue(xmlString, RSS.class);
+    Channel channel = rssFeed.getChannel();
+    channel.addItem(episodeTitle, episodeDescription, episodeLanguage, email, mp3Link);
```

Comment:
i think this will allow any user to modify any rss feed if they make a post request with someone else's rss feed. i think we'll need to do a email comparison before we actually add the item

Reply:
Makes sense!